### PR TITLE
Resolved the Installation via CMake also brings in 3rd party Dependency Issue

### PR DESCRIPTION
**Pull request by vatsalcode**
This PR addresses issue #13  where Google Benchmark was being installed by default, leading to unnecessary dependencies in production builds. By making the inclusion of Google Benchmark conditional, we reduce the build time and avoid installing unwanted dependencies in environments where benchmarks are not needed.

**Changes Made:**
- Added a CMake option `BUILD_BENCHMARKS` to control the inclusion of the benchmark subdirectory.
- Modified the root `CMakeLists.txt` to only fetch and build Google Benchmark when `BUILD_BENCHMARKS` is set to ON.

**Rationale:**
These changes allow developers and users to opt-in to benchmarking only when needed, streamlining the build process for production environments.

**Testing:**
- Compiled and ran the project with `BUILD_BENCHMARKS=OFF` and verified that Google Benchmark is not included.
- Compiled and ran with `BUILD_BENCHMARKS=ON` to ensure benchmarks still function as expected.

**Potential Impact:**
No impact on the main functionality of the project is expected. Benchmarks are entirely optional and are only included based on the build configuration.

Please review the changes and provide your feedback on the implementation of the conditional inclusion logic. Any insights or suggestions would be greatly appreciated!

-Vatsal

### CODE REVIEW

**Comment by wilderfield** [comment]
This is a good idea, my only thought is, along with this cleanup we could move the FetchContent down to the benchmark/CMakeLists.txt since that subfolder is where it’s needed.

**Comment by vatsalcode** [comment]
> This is a good idea, my only thought is, along with this cleanup we could move the FetchContent down to the benchmark/CMakeLists.txt since that subfolder is where it’s needed.

Thank you for the feedback! I agree that moving the FetchContent call for Google Benchmark to the benchmark/CMakeLists.txt makes a lot of sense since it isolates the dependency to where it's actually needed. This change would not only clean up the main CMakeLists.txt but also keep the benchmark-specific configurations contained within their own scope, which enhances maintainability and clarity of the build scripts.

I'll make these adjustments and update the pull request accordingly. I'll also ensure to test the changes to confirm everything works as expected with the new configuration. Thanks again for the guidance.

File: CMakeLists.txt [hunk 1]
```diff
@@ -14,23 +14,16 @@ FetchContent_Declare(
...
-
-# Configure Google Benchmark
-# Set this to disable tests and other unnecessary parts of the benchmark library
-set(BENCHMARK_ENABLE_TESTING OFF CACHE BOOL "Disable benchmark testing as we don't need it.")
+# Option to build benchmarks
+option(BUILD_BENCHMARKS "Build benchmarks" OFF)
 
-FetchContent_MakeAvailable(benchmark)
+if(BUILD_BENCHMARKS)
+    # Add the benchmark directory
```
Comment (wilderfield) [inline CMakeLists.txt]: Nit: Comment is too obvious
```suggestion
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -39,16 +32,14 @@ include_directories(include)
...
-
 # Build documentation
 add_subdirectory(docs)
 
 include(GNUInstallDirs)
 
 # Create an interface target for header-only library
 add_library(${PROJECT_NAME} INTERFACE)
-target_include_directories(${PROJECT_NAME}
+target_include_directories(${PROJECT_NAME}
```
Comment (wilderfield) [inline CMakeLists.txt]:
```suggestion
target_include_directories(${PROJECT_NAME}
```

**Review by wilderfield — APPROVED** [review]

**Pull request merged**